src/layout.ts

```diff
@@ -48,17 +48,43 @@ export function layoutPage(content: Paragraph[], box: Box, fonts: Font[]): Frame
 function layoutParagraph(content: Paragraph, box: Box, fonts: Font[]): Frame {
   const maxWidth = box.width;
   const maxHeight = box.height;
-  const { text, ...attrs } = content;
-  const segments = extractTextSegments(text, attrs, fonts);
-  return layoutRow(segments, { ...box, width: maxWidth, height: maxHeight });
+  const innerBox = { x: 0, y: 0, width: maxWidth, height: maxHeight };
+  const rows = content.text && layoutText(content, innerBox, fonts);
+  return {
+    type: 'paragraph',
+    ...box,
+    width: rows?.size?.width ?? 0,
+    height: rows?.size?.height ?? 0,
+    ...(rows?.rows?.length ? { children: rows.rows } : undefined),
+  };
+}
+
+function layoutText(content: Paragraph, box: Box, fonts: Font[]) {
```

Comment:
The name of the function is confusing for me. I could not understand what it was doing from reading the name.